### PR TITLE
platform: intel: ace: remove all unnecessary clk.h definitions

### PR DESCRIPTION
**Pull request by kv2019i**
Strip all unnecessary definitions from Intel ACE clk.h platform code. Only expose the main DSP clock via the SOF clk.h layer, as required by cpu-clk-manager.c.

### CODE REVIEW

File: src/platform/intel/ace/include/ace/lib/clk.h [hunk 1]
```diff
@@ -20,30 +20,16 @@
...
-#define CLK_CPU(x)	(x)
-
-/** \brief SSP clock r-t settings are after the core(s) settings */
-#define CLK_SSP		CONFIG_CORE_COUNT
-
-/* SSP clock run-time data is the last one, so total number is ssp idx +1 */
-
 /** \brief Total number of clocks */
-#define NUM_CLOCKS	(CLK_SSP + 1)
+#define NUM_CLOCKS	NUM_CPU_FREQ
```
Comment (lyakh) [inline src/platform/intel/ace/include/ace/lib/clk.h:27]: this is a bit confusing. It's used in https://github.com/thesofproject/sof/blob/5389b257519c1fa00873386d94b70825dee04271/zephyr/lib/clk.c#L16 as a size of array, but then lines https://github.com/thesofproject/sof/blob/5389b257519c1fa00873386d94b70825dee04271/zephyr/lib/clk.c#L29-L32 iterate over that array, but using `CONFIG_CORE_COUNT` as the iteration number?

Comment (kv2019i) [reply src/platform/intel/ace/include/ace/lib/clk.h]: Thanks @lyakh , this was a good catch. This is mixing number of clocks and number of frequencies (each CPU clock has). Fixed version uploaded now.

File: zephyr/lib/clk.c [hunk 2]
```diff
@@ -28,6 +28,8 @@ void platform_clock_init(struct sof *sof)
 
 	sof->clocks = platform_clocks_info;
 
+	BUILD_ASSERT(NUM_CLOCKS >= CONFIG_CORE_COUNT, "Invalid NUM_CLOCKS");
```
Comment (lgirdwood) [inline zephyr/lib/clk.c:31]: Clock/cores ?

**Review by lgirdwood — COMMENTED** [review]
LGTM, just one question.

File: zephyr/lib/clk.c (hunk 2 above)
Comment (lgirdwood) [reply zephyr/lib/clk.c:31]: I would add an inline comment as this may create confusion.

**Review by lgirdwood — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Pull request merged**